frmts/mem/memdataset.h

```diff
@@ -135,6 +135,9 @@ class CPL_DLL MEMDataset CPL_NON_FINAL : public GDALDataset
 
     OGRLayer *GetLayer(int) override;
 
+    OGRLayer *CreateLayer(const OGRFeatureDefn &oDefn,
```

Comment:
if you return a OGRMemLayer* you could save the cpl::down_cast<> in gdalalg_vector_geom_simplify_coverage.cpp

Reply:
Good idea, thanks